Client/Frontend/Autofill/CreditCard/ViewComponents/CreditCardInput/CreditCardInputViewModel.swift

```diff
@@ -0,0 +1,95 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import Foundation
+
+class CreditCardInputViewModel: ObservableObject {
+    let fieldType: CreditCardInputType
+    let fieldHeadline: String
+    let errorString: String
+    var formattedTextLimit: Int
+    var delimiterCharacter: String?
+    var keyboardType: UIKeyboardType
+
+    @Published var isValid = true
+
+    init(fieldType: CreditCardInputType) {
+        self.fieldType = fieldType
+
+        switch fieldType {
+        case .name:
+            self.fieldHeadline = .CreditCard.EditCard.NameOnCardTitle
+            self.errorString = .CreditCard.ErrorState.NameOnCardSublabel
+            self.delimiterCharacter = nil
+            self.formattedTextLimit = 100 // TBA
+            self.keyboardType = .alphabet
+        case .number:
+            self.fieldHeadline = .CreditCard.EditCard.CardNumberTitle
+            self.errorString = .CreditCard.ErrorState.CardNumberSublabel
+            self.formattedTextLimit = 20
+            self.delimiterCharacter = " "
+            self.keyboardType = .numberPad
+        case .expiration:
+            self.fieldHeadline = .CreditCard.EditCard.CardExpirationDateTitle
+            self.errorString = .CreditCard.ErrorState.CardExpirationDateSublabel
+            self.formattedTextLimit = 7
+            self.delimiterCharacter = " / "
+            self.keyboardType = .numberPad
+        case .securityCode: // Design feedback needed
+            self.fieldHeadline = "Security Code"
```

Comment:
These two strings need to be localized.

Reply:
The security code is something I need to ask design about. Maybe excluding it was intentional, and if it was, then I'll remove all securityCode related code. Should be updated EOD.

Reply:
Updating to remove all security code related items.